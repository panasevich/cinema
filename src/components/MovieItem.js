import React from 'react';
import PropTypes from 'prop-types';
import { Link, Route} from 'react-router-dom';
import { Card, CardImg, CardBody,
    CardTitle, Col, CardSubtitle, Button } from 'reactstrap';

export default function MovieItem(props) {
    const {imgPath, title, genre, year, movieId} = props;
    return(
        <Col sm="4" style={{margin: '15px 0'}}>
            <Card>
                <CardImg top src={imgPath} alt="Card image cap" />
                <CardBody>
                    <CardTitle>{title}</CardTitle>
                    <CardSubtitle>{year}</CardSubtitle>
                    {genre.map((item, idx) => <div key={`${item}-${idx}`}>{item}</div>)}
                    <Button><Link to={`/movie/${movieId}`}>View</Link></Button>
                </CardBody>
            </Card>
        </Col>
    )
}

MovieItem.propTypes = {
    imgPath: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    year: PropTypes.number.isRequired,
    genre: PropTypes.array.isRequired,
};

