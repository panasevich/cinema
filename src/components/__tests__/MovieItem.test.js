import React from 'react';
import MovieItem from '../MovieItem';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('Movie item tests', () => {
    it('renders correctly', () => {
        const imgPath = 'url';
        const title = 'Title';
        const year = '2005';
        const genre = ['action', 'romance'];
        const tree = renderer
            .create(
                <MemoryRouter>
                    <MovieItem
                        title={title}
                        year={year}
                        genre={genre}
                        imgPath={imgPath}
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});