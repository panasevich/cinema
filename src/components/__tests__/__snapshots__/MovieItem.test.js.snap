// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie item tests renders correctly 1`] = `
<div
  className="col-sm-4"
  style={
    Object {
      "margin": "15px 0",
    }
  }
>
  <div
    className="card"
  >
    <img
      alt="Card image cap"
      className="card-img-top"
      src="url"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Title
      </h5>
      <h6
        className="card-subtitle"
      >
        2005
      </h6>
      <div>
        action
      </div>
      <div>
        romance
      </div>
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        <a
          href="/movie/undefined"
          onClick={[Function]}
        >
          View
        </a>
      </button>
    </div>
  </div>
</div>
`;
